### PR TITLE
Fix $ conflicts

### PR DESCRIPTION
**Pull request by Yago**
### 💬 Description

~~This was a tricky one...~~

~~First, the bug was an error on [preprod-mon-espace.t-l.ch/](https://preprod-mon-espace.t-l.ch/), a site that we didn't produce…~~
```
leckerli.min.js:17 Uncaught TypeError: Method RegExp.prototype.exec called on incompatible receiver #<HTMLInputElement>
    at HTMLInputElement.test (<anonymous>)
    at leckerli.min.js:17
    at HTMLInputElement.<anonymous> (newsletter.js?v=38:36)
    at HTMLInputElement.dispatch (jquery-3.6.3.min.js?v=38:2)
    at HTMLInputElement.y.handle (jquery-3.6.3.min.js?v=38:2)
```

~~After a quick update of our dependencies, the error changes to~~

```
newsletter.js?v=38:36 Uncaught TypeError: $(...).data is not a function
    at HTMLInputElement.<anonymous> (newsletter.js?v=38:36:45)
    at HTMLInputElement.dispatch (jquery-3.6.3.min.js?v=38:2:43336)
    at y.handle (jquery-3.6.3.min.js?v=38:2:41320)
```

~~And fixed! Right? Because no more `leckerli` relation in the error. Sadly, it was not so easy, because without Leckerli, no error occurs...~~

~~After a bit of digging, it seems that [DOMPurify](https://github.com/cure53/DOMPurify) is the source of the issue; it alters the nature of `$` and “disable” some of jQuery available methods. So without DOMPurify and its safety, no more bug? Almost, `$`, was still overridden by something in our bundle...~~

~~So instead of removing another lib, trying to run everything in a Web Worker (still not a bad idea) or rewriting the entire application, I use a small hack that seems to work just fine 😇~~

Change build system for `iife` format and more logic scoping.

### 🔢 To Review
1.  Try Leckerli with and without jQuery

- [x] Update the "Unreleased" section of the `CHANGELOG.md` with [chan](https://github.com/geut/chan/tree/main/packages/chan)

### CODE REVIEW

**Comment by gido** [comment]
I think we have a larger issues. I quickly checked what is the "impact" on the window object when we load our script.

I found out that "we" added more than 200 variables and functions to the global window scope.

<details>
<summary>see 203 variables / functions added</summary>

```
Propriétés ajoutées (203 éléments):

    Fn
    xe
    wn
    yi
    gi
    kt
    Qn
    _i
    Hn
    ze
    Si
    Wt
    Ti
    Ir
    Ai
    zr
    ri
    Ko
    xo
    Te
    Sn
    Jn
    Bo
    An
    Ji
    Kr
    Br
    $o
    ae
    ti
    ki
    Io
    zo
    Ve
    mr
    Zi
    qo
    Un
    Ln
    Ei
    kn
    bn
    d
    Zo
    ht
    Wn
    En
    Xr
    mi
    ee
    bi
    Mo
    fi
    T
    er
    Ze
    Pn
    xn
    Yo
    hi
    Qr
    Fo
    wo
    $n
    Rn
    jr
    zn
    Sr
    Jr
    hn
    Ar
    wt
    Ho
    yo
    be
    et
    Tn
    Xo
    Ur
    jn
    ui
    tr
    li
    qr
    ci
    Ro
    Dn
    Nr
    Uo
    ni
    nr
    ko
    On
    J
    rr
    Y
    Ne
    or
    X
    B
    Re
    ir
    Ot
    qe
    ar
    sr
    Q
    lr
    ur
    vo
    yn
    Ye
    vn
    gn
    Mt
    cr
    go
    _r
    fr
    Eo
    Ee
    pr
    bo
    dr
    le
    De
    Pt
    Je
    Le
    hr
    yr
    Qe
    vr
    gr
    Ut
    To
    Er
    Ao
    So
    Be
    Ct
    $t
    br
    He
    Nt
    O
    U
    te
    Tr
    Cn
    Nn
    Co
    ne
    No
    In
    Ft
    Rt
    Do
    Lo
    kr
    xt
    vt
    Xe
    Or
    Cr
    wr
    Rr
    we
    xr
    Dr
    ge
    Go
    Vo
    Lr
    Mr
    Pr
    Jo
    Qo
    ei
    $r
    jt
    Fr
    Hr
    Wr
    zt
    Gr
    Vr
    Yr
    ii
    ai
    si
    pi
    gt
    di
    vi
    Ni
    H
    Fi
    g
    ve
    We
    ro
    A
```
</details>


<details>
<summary>Simple HTML Code to execute the test</summary>

```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Test de Pollution de l'objet Window</title>
    <link rel="stylesheet" href="assets/leckerli.min.css" />
</head>
<body>

<script>
    let initialWindowProperties = Object.getOwnPropertyNames(window);

    function compareWindowProperties() {
        let afterLibraryLoadProperties = Object.getOwnPropertyNames(window);
        
        let addedProperties = afterLibraryLoadProperties.filter(prop => !initialWindowProperties.includes(prop));

        console.log('Propriétés ajoutées:', addedProperties);

        // Display as an HTML list.
        if (addedProperties.length) {
            let title = document.createElement('h2');
            title.textContent = `Propriétés ajoutées (${addedProperties.length} éléments):`;
            document.body.appendChild(title);

            let ul = document.createElement('ul');

            addedProperties.forEach(prop => {
                let li = document.createElement('li');
                let codeElement = document.createElement('code');
                codeElement.textContent = prop;
                li.appendChild(codeElement);
                ul.appendChild(li);
            });

            document.body.appendChild(ul);
        }
    }

</script>

<script src="https://www.unpkg.com/@antistatique/leckerli@1.2.0/dist/assets/leckerli.min.js" onload="compareWindowProperties()"></script>
</body>
</html>
```
</details>

We need to find a way to drastically reduce our "footprints" to avoid any other conflicts.

Is Vite / Rollup really the good candidate to manage our build?

Also, in this test, I didn't find the `window.$` conflict.

/cc @yannlugrin @Yago

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Fixed
+- Fix jQuery conflicts
```
Comment (gido) [inline CHANGELOG.md]:
```suggestion
- Change build system for iife format and more logic scoping (fix "jQuery conflicts")
```

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    compact: true,
+    sourcemap: false,
+    format: 'iife',
+    dir: './dist/assets',
+    entryFileNames: name + '.min.js',
+    inlineDynamicImports: true,
+  },
+  plugins: [
+    replace({
+      'process.env.NODE_ENV': JSON.stringify('production'),
```
Comment (gido) [inline rollup.config.js:25]: We also need to replace `APP_VERSION`.

see https://github.com/antistatique/leckerli/blob/78d0d1b3ad1b314b4a85fdc9e15e31be250e9b82/vite.config.ts#L23 (required for the GTM autoload feature if I remember correctly)

I don't really like to have different config / system for the development and for the production build.

**Review by gido — CHANGES_REQUESTED** [review]

File: rollup.config.js (hunk 2 above)
Comment (Yago) [reply rollup.config.js:25]: Done; I add the missing `APP_VERSION` and I remove Vite from the project to keep only Rollup for the JS, Tailwind for CSS and BrowserSync to serve HTML during dev. Still, I'm not entirely satisfied to loose Vite DX... 🤔

File: README.md [hunk 3]
```diff
@@ -14,15 +14,15 @@ To use leckerli into your website, simply add the following CSS/JavaScript in yo
   <!-- somewhere in the <head /> -->
   <link
     rel="stylesheet"
-    href="//www.unpkg.com/@antistatique/leckerli@1.1/dist/assets/leckerli.min.css"
+    href="//www.unpkg.com/@antistatique/leckerli@1.1/dist/leckerli.min.css"
```
Comment (gido) [inline README.md]: We can't change this url without releasing a major version. Existing GTM integration could fail.

**Review by gido — CHANGES_REQUESTED** [review]

**Review by gido — APPROVED** [review]

**Pull request merged**